### PR TITLE
htlcswitch+peer+lnwire: pull weeds

### PR DESCRIPTION
**Pull request by ProofOfKeags**
## Change Description
In the process of trying to hack dynamic commitments into the htlcswitch package I discovered some weeds that I decided to pull. A lot of these commits stand alone but I fully recommend we make all of these changes.

## Steps to Test
All of these should be semantics invariant, so existing tests should be sufficient.

### Testing
- [x] Your PR passes all CI checks.
- [x] Tests covering the positive and negative (error paths) are included.
- [x] Bug fixes contain tests triggering the bug to prevent regressions.

### Code Style and Documentation
- [x] The change obeys the [Code Documentation and Commenting](https://github.com/lightningnetwork/lnd/blob/master/docs/code_contribution_guidelines.md#CodeDocumentation) guidelines, and lines wrap at 80.
- [x] Commits follow the [Ideal Git Commit Structure](https://github.com/lightningnetwork/lnd/blob/master/docs/code_contribution_guidelines.md#IdealGitCommitStructure).
- [x] Any new logging statements use an appropriate subsystem and logging level.
- [x] Any new lncli commands have appropriate tags in the comments for the rpc in the proto file.
- [x]  [There is a change description in the release notes](https://github.com/lightningnetwork/lnd/tree/master/docs/release-notes), or `[skip ci]` in the commit message for small changes.

📝 Please see our  [Contribution Guidelines](https://github.com/lightningnetwork/lnd/blob/master/docs/code_contribution_guidelines.md) for further guidance.

## Summary by CodeRabbit

- **New Features**
	- Introduced a new `LinkUpdater` interface for BOLT 2 messages that can update the channel state.
- **Refactor**
	- Modified the `Peer()` method across various interfaces and implementations to return the serialized public key instead of the full peer object.
	- Simplified the handling of peer public keys in the `Switch` struct by using direct references.
	- Adjusted test setups and variable naming to align with the changes in peer information handling and channel link configurations.
- **Chores**
	- Removed obsolete references and assignments related to `Switch` in channel configurations and tests.

### CODE REVIEW

File: htlcswitch/link.go [hunk 1]
```diff
@@ -92,13 +92,6 @@ type ChannelLinkConfig struct {
 	// allowing the link to open and close circuits.
 	Circuits CircuitModifier
 
-	// Switch provides a reference to the HTLC switch, we only use this in
-	// testing to access circuit operations not typically exposed by the
-	// CircuitModifier.
-	//
-	// TODO(conner): remove after refactoring htlcswitch testing framework.
-	Switch *Switch
```
Comment (Roasbeef) [inline htlcswitch/link.go:100]: 👍 

was long left over, nice to finally remove

File: htlcswitch/link_test.go [hunk 2]
```diff
@@ -319,7 +320,7 @@ func TestChannelLinkSigThenRev(t *testing.T) {
 
 	const chanAmt = btcutil.SatoshiPerBitcoin * 5
 	const chanReserve = btcutil.SatoshiPerBitcoin * 1
-	aliceLink, bobChannel, batchTicker, start, restore, err :=
+	aliceSwitch, aliceLink, bobChannel, batchTicker, start, restore, err :=
```
Comment (Roasbeef) [inline htlcswitch/link_test.go:323]: At this point, perhaps we should morph the harness into an actual struct so we don't have 6+ return values at the call site?

File: htlcswitch/interfaces.go [hunk 3]
```diff
@@ -262,9 +261,9 @@ type ChannelLink interface {
 	// total sent/received milli-satoshis.
 	Stats() (uint64, lnwire.MilliSatoshi, lnwire.MilliSatoshi)
 
-	// Peer returns the representation of remote peer with which we have
-	// the channel link opened.
-	Peer() lnpeer.Peer
+	// Peer returns the serialized public key of remote peer with which we
+	// have the channel link opened.
+	Peer() [33]byte
```
Comment (Roasbeef) [inline htlcswitch/interfaces.go:266]: `PeerPubkey()`?

**Review by Roasbeef — APPROVED** [review]
LGTM ☄️

File: htlcswitch/link_test.go (hunk 2 above)
Comment (ProofOfKeags) [reply htlcswitch/link_test.go]: Addressed.

File: htlcswitch/interfaces.go (hunk 3 above)
Comment (ProofOfKeags) [reply htlcswitch/interfaces.go]: Addressed.

File: htlcswitch/link_test.go [hunk 4]
```diff
@@ -187,28 +187,30 @@ func TestChannelLinkRevThenSig(t *testing.T) {
...
-	alice := newPersistentLinkHarness(
-		t, aliceLink, batchTicker, restore,
+	alice := newPersistentLinkHarness(t, harness.aliceSwitch,
+		harness.aliceLink, harness.aliceBatchTicker,
+		harness.aliceRestore,
 	)
 
 	var (
-		coreLink  = aliceLink.(*channelLink)
+		//nolint: forcetypeassert
```
Comment (ziggie1984) [inline htlcswitch/link_test.go:203]: Q: I wonder whether we miss some functions in the ChannelLink interface forcing us to type convert ?

**Review by ziggie1984 — APPROVED** [review]
Nice cleanup, easy to follow commit structure  👌

File: htlcswitch/link_test.go (hunk 4 above)
Comment (ProofOfKeags) [reply htlcswitch/link_test.go:203]: Definitely. I think our practice of breaking the glass on the interface in order to test is definitely a problem. It's just one I have a much longer story arc on solving. I'm aligned on this though. We shouldn't be doing it and I'm actively trying to dramatically clean up the htlcswitch package, but it's very delicate and there will be a lot of incremental steps towards getting us to a point where we can successfully deal with this.

**Pull request merged**